.github/workflows/build.yml

```diff
@@ -0,0 +1,188 @@
+on:
+  workflow_call:
+    inputs:
+      host-platform:
+        required: true
+        type: string
+      cuda-version:
+        type: string
+        default: "12.8.0"
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -xeuo pipefail {0}
+
+permissions:
+  contents: read  # This is required for actions/checkout
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.13"
+    name: py${{ matrix.python-version }}
+    runs-on: ${{ (inputs.host-platform == 'linux-64' && 'linux-amd64-cpu8') ||
+                 (inputs.host-platform == 'linux-aarch64' && 'linux-arm64-cpu8') ||
+                 (inputs.host-platform == 'win-64' && 'windows-2019') }}
+    steps:
+      - name: Checkout ${{ github.event.repository.name }}
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup proxy cache
+        uses: nv-gha-runners/setup-proxy-cache@main
+        continue-on-error: true
+        # Skip the cache on Windows nodes outside of our org.
+        if: ${{ inputs.host-platform != 'win-64' }}
+
+      - name: Set up Python
+        if: ${{ startsWith(inputs.host-platform, 'linux') }}
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
```

Comment:
Does this need to use the python version matrix?

Reply:
It was not, and does not seem to need to. Good call out as I had meant to ask and forgotten - @leofang https://github.com/NVIDIA/cuda-python/blob/main/.github/workflows/build-and-test.yml#L56-L61 is where I got this from and it was said the matrix version there, but then set up 3.12 everytime.

Reply:
@cryos Back then there was a nasty setup-python bug (see the comment right above the lines you linked to) that made the build steps fail. That was why the old build pipeline needed to lock to PY312. But when actually building the wheels (using the cibuildwheel action), we honor the Python version set in the matrix. I haven't checked the latest new build pipeline carefully -- will try to get to it tomorrow.